### ISSUE TITLE
Module not found: Error: Can't resolve '@mdx-js/react'

### ISSUE THREAD

**Issue by meriturva** [storybookjs/storybook#18599]
**Describe the bug**
there is no way to make Storybook work with the latest version of angular (14.0.x) after restoring packages.

**To Reproduce**

- ng new projectName
- cd projectName
- npx sb init
- npm run storybook (evviva! it works!)
- delete `package-lock.json` file and `node_modules` folder
- npm i
- npm run storybook (doh! error: Module not found: Error: Can't resolve '@mdx-js/react')

I have tried to use new MDXv2 as described here: https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#opt-in-mdx2-support (here issue related: #18094 ) without success: same error.

Any advice?

**Comment by meriturva** [comment]
I have found that you have to install both packages to make it work with MDXv2:

`npm install -D @storybook/mdx2-csf @mdx-js/react`

so basically it is better to update doc page with the correct package list.

**Comment by rs-pascal-lohscheidt** [comment]
Not specifically for Angular but...
For Web-Components, I was able to get it to work by forcing the `@mdx-js/react` package to be *1.6.22*.  Also I pinned down react to 17.x. 
All that has to be done is.
- Delete package-lock.json
- Add to your package.json
```json
"overrides": {
        "react": "^17.0.0",
        "react-dom": "^17.0.0",
        "@storybook/addon-docs": {
            "@mdx-js/react": "1.6.22"
        }
    }
```

**Comment by meriturva** [comment]
What I would like to point out is that issue is present on a new angular 14 project initialized with the command `npx sb init`. Now, our main configuration works again (using MDXv2), but it is important to fix it and keep Storybook friction to new developers lower as possible.

**Comment by agrauch** [comment]
This is also an issue for projects using React 18.

@meriturva fix works although the output, at least for [documentation only](https://storybook.js.org/docs/react/writing-docs/mdx#documentation-only-mdx) MDX, looks terrible.

**Comment by ndelangen** [comment]
This should be fixed in 7.0 beta

**Comment by cbregantejs** [comment]
I solved it by adding these dependencies and @mdx-js/react in overrides to avoid using --legacy-peer-deps
```
{
  "dependencies": {
    "react": "18.2.0",
  },
  "devDependencies": {
    "@mdx-js/react": "^1.6.22",
  },
  "overrides": {
    "@mdx-js/react": {
      "react": "$react"
    }
  }
}
```
